app/models/product.rb

```diff
@@ -15,4 +15,8 @@ def current_price
     a = price_histories.order(:date).limit(1).first
     a ? a.price : nil
   end
+
+  def search
+    raise NotImplementedError, 'Need to Implement on subClass'
+  end
```

Comment:
Your subclass implements this as a class method (`self.search`), which I would recommend. The parent class should _also_ implement as a class method, then.